### PR TITLE
Fixed several issues in the calculation of nucleation tendencies

### PR DESCRIPTION
**Pull request by jeff-cohere**
These changes allow our eamxx-mam4xx column test to compare closely with the MAM4 box model's ["smoke test"](https://github.com/eagles-project/mam_refactor/blob/main/standalone/tests/smoke_test.nl). However, they seem to cause a failure in our [amicphys unit test](https://github.com/eagles-project/mam4xx/blob/main/src/tests/mam4_amicphys_1gridcell.cpp).

@overfelt , I'm curious if you have any thoughts about this other than what we've (briefly) discussed. I'm fairly confident that these fixes to the nucleation parameterization are needed, but I don't know if I've missed something that might also be necessary for the amicphys test to pass.

After this, we can focus more on the simulation described in #169

### CODE REVIEW

**Comment by jeff-cohere** [comment]
EAMxx nucleation test results (H2SO4 gas nucleating to sulfate aerosol):

mam4xx_nucleation_time_series

**Comment by jeff-cohere** [comment]
Here's what's failing:

```
/home/runner/work/mam4xx/mam4xx/src/tests/mam4_amicphys_1gridcell.cpp:1946qgas4[1] != Approx(check_qgas4[1])): 1.9185977459535e-11 != 1.9012166840859e-11 with scale:1.9012166840859e-11 and epsilon:0.0001
```

File: src/mam4xx/nucleation.hpp [hunk 1]
```diff
@@ -698,7 +698,7 @@ class Nucleation {
           Real pblh = atm.planetary_boundary_layer_height;
           Real qv = atm.vapor_mixing_ratio(k);
           Real relhum = conversions::relative_humidity_from_vapor_mixing_ratio(
-              qv, pmid, temp);
+              qv, temp, pmid);
```
Comment (jeff-cohere) [inline src/mam4xx/nucleation.hpp:701]: !!!!!

**Comment by jeff-cohere** [comment]
[This line](https://github.com/eagles-project/mam4xx/pull/141/files#r1143747027) in #141 is likely related.

**Comment by jeff-cohere** [comment]
@overfelt and @pbosler , this PR is ready for action. With it merged, we can focus on our low-resolution global simulation of nucleating aerosols.

**Review by overfelt — APPROVED** [review]

**Pull request merged**